sklearn/model_selection/tests/test_validation.py

```diff
@@ -740,3 +740,37 @@ def test_cross_val_predict_sparse_prediction():
     preds_sparse = cross_val_predict(classif, X_sparse, y_sparse, cv=10)
     preds_sparse = preds_sparse.toarray()
     assert_array_almost_equal(preds_sparse, preds)
+
+
+def test_cross_val_predict_apply_func_cv():
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    classes = len(set(y))
+
+    # Shuffle the data before doing cross validation
+    shf = np.random.permutation(np.arange(len(X)))
+    X = X[shf]
+    y = y[shf]
+
+    kfold = KFold(len(iris.target))
+
+    func_names = ['decision_function', 'predict_proba', 'predict_log_proba']
+    for func_name in func_names:
+        cv_generator = kfold.split(X, y)
+
+        est = LogisticRegression()
+
+        predictions = cross_val_predict(est, X, y, apply_func=func_name)
+        assert_equal(len(predictions), len(y))
+
+        expected_predictions = np.zeros([len(y), classes])
+        func = getattr(est, func_name)
+
+        # Naive loop (should be same as cross_val_predict):
+        for train, test in cv_generator:
```

Comment:
please replace `cv_generator` by `kfold.split(X, y)` here and delete the equivalent above.